backend/open_webui/utils/telemetry/metrics.py

```
@@ -0,0 +1,110 @@
+        description="Total HTTP requests",
+        unit="1",
+    )
+    duration_histogram = meter.create_histogram(
+        name="http.server.duration",
+        description="HTTP request duration",
+        unit="ms",
+    )
+
+    # FastAPI middleware
+    @app.middleware("http")
+    async def _metrics_middleware(request: Request, call_next):
```

Ruff (0.11.9)

Missing return type annotation for private function `_metrics_middleware`

(ANN202)